### PR TITLE
feat: add unit tests in C++

### PR DESCRIPTION
**Pull request by Stitch07**

### CODE REVIEW

File: test.cc [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#include <iostream>
+
+int main()
+{
+    std::cout << "Hello world!\n";
```
Comment (Tylertron1998) [inline test.cc:5]: why is this here?

Comment (Stitch07) [reply test.cc:5]: whoops

File: binding.gyp [hunk 2]
```diff
@@ -2,6 +2,9 @@
   "targets": [
     {
       "target_name": "skyra-ai",
+      "dependencies": [
+        "deps/binding.gyp:googletest"
+      ],
```
Comment (kyranet) [inline binding.gyp:7]: Does it need googletest in the production build?

File: tests/Connect4.cc [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#include "gtest/gtest.h"
+#include "games/ConnectFour.hpp"
+
+TEST(Connect4, Stub) {
+	connect_four::ai_board b{};
+	EXPECT_EQ(connect_four::position(b, 42, 5), 3);
+}
```
Comment (kyranet) [inline tests/Connect4.cc:7]:
```suggestion
	connect_four::ai_board b{};
	EXPECT_EQ(connect_four::position(b, 42, 5), 3);
}
```

**Review by kyranet — CHANGES_REQUESTED** [review]

File: binding.gyp (hunk 2 above)
Comment (Stitch07) [reply binding.gyp:7]: Resolved in b70c2b0

**Review by kyranet — APPROVED** [review]

**Pull request merged**